src/main/java/com/iridium/iridiumskyblock/commands/CommandManager.java

```diff
@@ -6,74 +6,123 @@
+     * All parameters are provided by Bukkit.
+     *
+     * @param commandSender The sender which executes this command
+     * @param cmd The Bukkit {@link org.bukkit.command.Command} representation
+     * @param label The label of this command. Not used.
+     * @param args The arguments of this command
+     * @return true if this command was executed successfully
+     */
     @Override
-    public boolean onCommand(CommandSender cs, org.bukkit.command.Command cmd, String s, String[] args) {
-        if (args.length != 0) {
-            for (Command command : commands) {
-                if (command.aliases.contains(args[0]) && command.enabled) {
-                    if (command.player && !(cs instanceof Player)) {
-                        // Must be a player
-                        cs.sendMessage(StringUtils.color(iridiumSkyblock.getMessages().mustBeAPlayer
-                                .replace("%prefix%", iridiumSkyblock.getConfiguration().prefix)));
-                        return true;
-                    }
-                    if ((cs.hasPermission(command.permission) || command.permission
-                            .equalsIgnoreCase("") || command.permission
-                            .equalsIgnoreCase("iridiumskyblock.")) && command.enabled) {
-                        command.execute(cs, args);
-                    } else {
-                        // No permission
-                        cs.sendMessage(StringUtils.color(iridiumSkyblock.getMessages().noPermission
-                                .replace("%prefix%", iridiumSkyblock.getConfiguration().prefix)));
-                    }
-                    return true;
-                }
-            }
-        } else {
-            if (cs instanceof Player) {
-                Player p = (Player) cs;
+    public boolean onCommand(@NotNull CommandSender commandSender, org.bukkit.command.@NotNull Command cmd, @NotNull String label, String[] args) {
+        if (args.length == 0) {
+            if (commandSender instanceof Player) {
+                Player p = (Player) commandSender;
+                // TODO: Send message
                 return true;
             }
         }
-        cs.sendMessage(StringUtils.color(iridiumSkyblock.getMessages().unknownCommand
+
+        for (Command command : commands) {
+            // We don't want to execute other commands or ones that are disabled
+            if (!(command.aliases.contains(args[0]) && command.enabled)) {
+                continue;
+            }
+
+            if (command.player && !(commandSender instanceof Player)) {
+                // Must be a player
+                commandSender.sendMessage(StringUtils.color(iridiumSkyblock.getMessages().mustBeAPlayer
+                        .replace("%prefix%", iridiumSkyblock.getConfiguration().prefix)));
+                return false;
+            }
+
+            if (!((commandSender.hasPermission(command.permission) || command.permission
```

Comment:
if the command.permission is empty then there is no permission required. so it should be && !command.permission.equals("")

Reply:
I just inverted the whole statement, it should work